sympy/physics/mechanics/body.py

```diff
@@ -0,0 +1,167 @@
+    >>> frame = ReferenceFrame('frame')
+    >>> body_inertia = inertia(frame, 1, 0, 0)
+    >>> body = Body('name_of_body', masscenter, mass, frame, body_inertia)
+
+    3. Creating a Particle. If masscenter and mass are passed, and inertia is
+     not then a Particle is created.
+
+    >>> from sympy import Symbol
+    >>> from sympy import Point
+    >>> from sympy.physics.mechanics import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Point('masscenter')
+    >>> body = Body('name_of_body', masscenter, mass)
+
+    Similarly, A frame can also be passed while creating a Particle.
+
+    """
+    def __init__(self, name, masscenter=None, mass=None, frame=None,
+                 body_inertia=None):
+
+        self.parent = None
+        self.child = None
+        self.force_list = []
+        self._counter = 0
+
+        if masscenter is None:
+            self._masscenter = Point(name + '_masscenter')
+        else:
+            self._masscenter = masscenter
+
+        if mass is None:
+            _mass = Symbol(name + '_mass')
+        else:
+            _mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(name + '_frame')
+        else:
+            self._frame = frame
+
+        if body_inertia is None and mass is None:
+            _inertia = (inertia(self._frame, 1, 1, 1), self._masscenter)
+        else:
+            _inertia = (body_inertia, self._masscenter)
+
+        self._masscenter.set_vel(self._frame, 0)
+
+        # If user passes masscenter and mass then a particle is created
+        # otherwise a rigidbody. As a result a body may or may not have inertia.
+        if body_inertia is None and mass is not None:
+            Particle.__init__(self, name, self._masscenter, _mass)
+        else:
+            RigidBody.__init__(self, name, self._masscenter, self._frame, _mass, _inertia)
+
+    def add_force(self, force_vector, point=None, frame=None):
+        """
+        Adds force to the body by adding a force vector in form of a tuple
+        of length 3 and a point on which the force will be applied.
+
+        Parameters
+        ---------
+        force_vector: A 3-Tuple
+            Defines the force vector w.r.t frame passed (or Body's frame). The
+            tuple defines the values of vector in x, y and z directions of the
+            frame.
+        point: Point
+            Defines the point on which the force must be applied.
+        frame: ReferenceFrame
+            Defines the frame w.r.t which force vector must be defined.
+
+        Example
+        --------
+        To add a force of magnitude 1 in y direction on a point at distance of
+        1 in x direction. All the directions are w.r.t body's frame.
+
+        >>> from sympy import Symbol
+        >>> from sympy.physics.mechanics import Body
+        >>> body = Body('body')
+        >>> g = Symbol('g')
+        >>> body.add_force((body.get_mass() * g, 0, 0), body.get_masscenter())
```

Comment:
That's great but RigidBody was badly designed in this respect. There is no reason to copy the bad parts. Maybe this is the time to modify Particle and RigidBody to have attributes with `@property`, you can leave the get/set methods in place for backwards compatibility. The new Body class should follow a better design.